clients/src/main/java/org/apache/kafka/common/requests/ListConfigResourcesRequest.java

```diff
@@ -28,14 +28,25 @@
 public class ListConfigResourcesRequest extends AbstractRequest {
     public static class Builder extends AbstractRequest.Builder<ListConfigResourcesRequest> {
         public final ListConfigResourcesRequestData data;
+        private final boolean fallbackToV0;
 
         public Builder(ListConfigResourcesRequestData data) {
             super(ApiKeys.LIST_CONFIG_RESOURCES);
             this.data = data;
+            this.fallbackToV0 = false;
+        }
+
+        public Builder(ListConfigResourcesRequestData data, boolean fallbackToV0) {
+            super(ApiKeys.LIST_CONFIG_RESOURCES);
+            this.data = data;
+            this.fallbackToV0 = fallbackToV0;
         }
 
         @Override
         public ListConfigResourcesRequest build(short version) {
+            if (fallbackToV0) {
```

Comment:
I suggest managing the fallback case slightly differently.

If the broker only supports v0, the build request will supply 0 as the parameter. Then you can reject the build request with `UnsupportedVersionException` if the config resource types are not just `CLIENT_METRICS`. I suggest looking at `ListGroupsRequest`.